File: Day225_2025/Q225_GFG.py
class Solution:
    def minSoldiers(self, arr, k):
        n = len(arr)
        safe_troop = (n + 1) // 2
        for i in range(n):
            arr[i] = (k - arr[i] % k) if arr[i] % k != 0 else 0
        arr.sort()
        ans = 0
        i = 0
        while safe_troop > 0:
            ans += arr[i]
            i += 1
            safe_troop -= 1
        return ans

File: Day225_2025/Q225_GFG.java
import java.util.Arrays;

class Q225_GFG {
    public int minSoldiers(int[] arr, int k) {
        int n = arr.length;
        int safe_troop = (n + 1) / 2;
        for (int i = 0; i < n; i++) {
            arr[i] = arr[i] % k != 0 ? k - arr[i] % k : 0;
        }
        Arrays.sort(arr);
        int i = 0, ans = 0;
        while (safe_troop > 0) {
            ans += arr[i];
            i++;
            safe_troop--;
        }
        return ans;
    }
}
